index.js

```diff
@@ -20,15 +20,19 @@ const getStream = (config) => {
       stream = { path: config.streamPath || './logs/app.log' };
       break;
     case 'SYSLOG':
+      let syslogStream = bsyslog.createBunyanStream({
+        name: config.streamName,
+        host: config.streamHost,
+        port: parseInt(config.streamPort, 10),
+        facility: bsyslog.facility.local0,
+        type: config.streamProto
+      });
+      if (config.streamProto.toUpperCase() === 'UDP') {
```

Comment:
can we cover both `udp` and `tcp`?

Reply:
TCPStream has it's own error handler

```js
        (function connect(event) {
                if (self.socket) {
                        if (self.listeners(event).length > 1) {
                                self.emit.apply(self, arguments);
                                return;
                        }

                        PROXY_EVENTS.forEach(function (e) {
                                self.socket.removeAllListeners(e);
                        });
                        self.socket.removeAllListeners('close');
                        self.socket.removeAllListeners('error');
                        if (self.socket.destroy) {
                                self.socket.destroy();
                        }
                }

                self.socket = createSocket({
                        host: self.host,
                        port: self.port,
                        proxy: self
                });
                self.socket.on('close', setTimeout.bind(null, connect, 1000));
                self.socket.on('error', setTimeout.bind(null, connect, 1000)); // <-- here
                self.socket.once('connect', function () {
                        self.queue.forEach(function (buf) {
                                self.socket.write(buf);
                        });
                });
        }());
```

And it won't emit error events when `_send`:

```js
TCPStream.prototype._send = function _send(msg) {
        this.socket.write(Buffer.from(msg + '\n', 'utf-8'));
};
```

So I think it's not necessary to change the default behavior of TCPStream, since it won't break our app even it encountered some error inside itself.

UDPStream is different, it would emit errors when `_send`, so it would throw uncaught exception to break our app:

```js
UDPStream.prototype._send = function _send(msg) {
        var buf = Buffer.from(msg, 'utf-8');
        var s = this.socket;
        var self = this;

        this._pending++;
        s.send(buf, 0, buf.length, this.port, this.host, function (err) {
                if (err) {
                        self.emit('error', err); // <-- here
                }

                self._pending--;
        });
};
```

Reply:
ok, glad to know